src/commands/PublishCommand.js

```diff
@@ -17,6 +17,11 @@ export default class PublishCommand extends Command {
       this.logger.info("Publishing canary build");
     }
 
+    const currentBranch = GitUtilities.getCurrentBranchDescription();
+    if (currentBranch && currentBranch.includes("detached at")) {
+      callback("You are working on detached branch, create new branch to publish changes.");
```

Comment:
We really shouldn't be [depending on the output](https://git-blame.blogspot.com/2013/06/checking-current-branch-programatically.html) of a porcelain command.

```bash
# correct example from the linked blog post
if branch=$(git symbolic-ref --short -q HEAD)
then
  echo on branch $branch
else
  echo not on any branch
fi
```

Determining if the branch is detached then becomes `if (currentBranch === '') {`